OSDOCS-19226 updated release notes for minor release 1.0.1 (#110523)
openshift/openshift-docs: modules/zero-trust-manager-release-notes-0-2-0.adoc

// Module included in the following assemblies:
//

> // * security/zero_trust_workload_identity_manager/zer-trust-manager-install.adoc

Reviewer: ```suggestion
// * security/zero_trust_workload_identity_manager/zero-trust-manager-install.adoc
```

:_mod-docs-content-type: REFERENCE
[id="zero-trust-manager-release-notes-0-2-0_{context}"]
= {zero-trust-full} 0.2.0 (General Availability)

[role="_abstract"]
Issued: 09 August 2025

> The following advisories are available for the {zero-trust-full}:

Reviewer 2: This needs a blank line here, otherwise the bullet list isn't rendering correctly in the preview:

```suggestion
The following advisories are available for the {zero-trust-full}:

```

Author: Done

* https://access.redhat.com/errata/RHBA-2025:15425[RHBA-2025:15425]
* https://access.redhat.com/errata/RHBA-2025:15426[RHBA-2025:15426]
* https://access.redhat.com/errata/RHBA-2025:15427[RHBA-2025:15427]
* https://access.redhat.com/errata/RHBA-2025:15428[RHBA-2025:15428]

> This release of {zero-trust-full} is a Technology Preview.

Reviewer 2: Same question here about a missing word at the end

Author: Removed because it is now GA

** `DefaultJWTValidity` (default: 5m)

* The default values can be replaced in the server configuration with user-configurable options that give users the flexibility to customize certificate and {svid-full} lifetimes based on their security requirements.

Enabling Manual User Configurations::

* The Operator controller switches to `create-only` mode once the `ztwim.openshift.io/create-only=true` annotation is present on the Operator's APIs. This allows resource creation while skipping the updates. A user can update the resources manually to test their configuration. This annotation supports APIs such as `SpireServer`, `SpireAgents`, `SpiffeCSIDriver`, `SpireOIDCDiscoveryProvider`, and `ZeroTrustWorkloadIdentityManager`.

* When the annotation is applied, all derived resources including resources created and managed by the Operator are created but not updated.

* After the annotation is removed and the pod restarts, the Operator tries to come back to the required state. The annotation is applied only once during start or a restart.

[id="zero-trust-manager-0-2-0-bug-fixes_{context}"]
== Fixed issues

JSON Web Token Issuer field now requires a valid URL::
+

> * Before this update, the `JwtIssuer` field for both the `SpireServer` and the `SpireOidcDiscoveryProvider` custom resources did not require the input to be a URL, frequently causing configuration errors. With this release, the validation has been updated, and users must now manually enter a valid issuer URL in the `JwtIssuer` field for both custom resources. As a result, misconfigurations caused by malformed issuer values are prevented, ensuring a stable and reliable setup.

Reviewer: ```suggestion
* Before this update, the `JwtIssuer` field for both the `SpireServer` and the `SpireOidcDiscoveryProvider` custom resources did not require the input to be a URL, frequently causing configuration errors. With this release, the validation has been updated, and users must now manually enter a valid issuer URL in the `JwtIssuer` field for both custom resources. As a result, misconfigurations caused by a malformed issuer values are prevented, ensuring a stable and reliable setup.
```

Author: DOne

// Module included in the following assemblies:
//
// * security/zero_trust_workload_identity_manager/zer-trust-manager-install.adoc
:_mod-docs-content-type: REFERENCE

> [id="zero-trust-manager-release-notes-0-2-0"]

Author: Done

= {zero-trust-full} 0.2.0 (Technology Preview)

[role="_abstract"]
Issued: 2025-09-08

> The following advisories are available for the {zero-trust-full}.

Reviewer: ```suggestion
The following advisories are available for the {zero-trust-full}:
```

Author: Done

Reviewer: .: Issue:

<img width="750" height="129" alt="Screenshot From 2026-05-19 14-22-02" src="https://github.com/user-attachments/assets/1e8015c7-9775-4466-960b-a5c8a2f760e0" />

Check the other 3 instances of the same line that should end in a colon.

Author: Done

* https://access.redhat.com/errata/RHBA-2025:15425[RHBA-2025:15425]
* https://access.redhat.com/errata/RHBA-2025:15426[RHBA-2025:15426]
* https://access.redhat.com/errata/RHBA-2025:15427[RHBA-2025:15427]
* https://access.redhat.com/errata/RHBA-2025:15428[RHBA-2025:15428]

This release of {zero-trust-full} is a Technology Preview.

[id="zero-trust-manager-0-2-0-features-enhancements_{context}"]
== New features and enhancements

Support for the managed OIDC Discovery Provider Route::

* The Operator exposes the `SPIREOIDCDiscoveryProvider` spec through {product-title} Routes under the domain `*.apps.<cluster_domain>` for the selected default installation.

> * The `managedRoute` and  `externalSecretRef` fields have been added to the `spireOidcDiscoveryProvider` spec.

Reviewer: ```suggestion
* The `managedRoute` and `externalSecretRef` fields have been added to the `spireOidcDiscoveryProvider` spec.
```

Author: Done

Enabling the custom Certificate Authority Time-To-Live for the SPIRE bundle::

* The following Time-To-Live (TTL) fields have been added to the `SpireServer` custom resource definition (CRD) API for SPIRE Server certificate management:

** `CAValidity` (default: 24h)

** `DefaultX509Validity` (default: 1h)

** `DefaultJWTValidity` (default: 5m)

* The default values can be replaced in the server configuration with user-configurable options that give users the flexibility to customize certificate and {svid-full} lifetimes based on their security requirements.

Enabling Manual User Configurations::

* The Operator controller switches to `create-only` mode once the `ztwim.openshift.io/create-only=true` annotation is present on the Operator's APIs. This allows resource creation while skipping the updates. A user can update the resources manually to test their configuration. This annotation supports APIs such as `SpireServer`, `SpireAgents`, `SpiffeCSIDriver`, `SpireOIDCDiscoveryProvider`, and `ZeroTrustWorkloadIdentityManager`.

> * When the annotation is applied, all derived resources including resources created and managed by the Operator.

Reviewer: Incomplete sentence. Maybe?

>  The annotation is applied to all derived resources including resources created and managed by the Operator

Author: Done. Made the sentence complete

> * Once the annotation is removed and the pod restarts, the operator tries to come back to the required state. The annotation is applied only once during start or a restart.

Reviewer: ```suggestion
* After the annotation is removed and the pod restarts, the Operator tries to return to the required state. The annotation is applied only once during start or a restart.
```

Author: Done

[id="zero-trust-manager-0-2-0-bug-fixes_{context}"]
== Fixed issues

JSON Web Token Issuer field now requires a valid URL::
+

> * Before this update, the `JwtIssuer` field for both the `SpireServer` and the `SpireOidcDiscoveryProvider` did not need to be a URL causing an error in configurations. With this release, the user must manually enter an issuer URL in the `JwtIssuer` field in both custom resources.

Reviewer: This bug does bot follow the CCFR pattern.

Author: Done
